flocker/control/test/test_httpapi.py

```diff
@@ -45,25 +72,344 @@ def assertGoodResult(self, method, path, expected_good_result):
+        assigned to an existing dataset on the indicated primary, the response
+        is an error indicating the collision and the dataset is not added to
+        the desired configuration.
+        """
+        return self._dataset_id_collision_test(self.NODE_A)
+
+    def test_unknown_primary_node(self):
+        """
+        If a ``POST`` request made to the endpoint indicates a non-existent
+        node as the location of the primary manifestation, the configuration is
+        unchanged and an error response is returned to the client.
+        """
+        return self.assertBadResult(
+            b"POST", b"/datasets", {u"primary": self.NODE_A},
+            BAD_REQUEST, {
+                u"description":
+                    u"The provided primary node is not part of the cluster."
+            }
+        )
+    test_unknown_primary_node.todo = (
+        "See FLOC-1278.  Make this pass by inspecting cluster state "
+        "instead of desired configuration to determine whether a node is "
+        "valid or not."
+    )
+
+    def test_minimal_create_dataset(self):
+        """
+        If a ``POST`` request made to the endpoint includes just the minimum
+        information necessary to create a new dataset (an identifier of the
+        node on which to place its primary manifestation) then the desired
+        configuration is updated to include a new unattached manifestation of
+        the new dataset with a newly generated dataset identifier and a
+        description of the new dataset is returned in a success response to the
+        client.
+        """
+        creating = self.assertResponseCode(
+            b"POST", b"/datasets", {u"primary": self.NODE_A},
+            OK)
+        creating.addCallback(readBody)
+        creating.addCallback(loads)
+
+        def got_result(result):
+            result = result[u"result"]
+            dataset_id = result.pop(u"dataset_id")
+            self.assertEqual(
+                {u"primary": self.NODE_A, u"metadata": {}}, result
+            )
+            deployment = self.persistence_service.get()
+            self.assertEqual({dataset_id}, set(get_dataset_ids(deployment)))
+        creating.addCallback(got_result)
+
+        return creating
+
+    def test_create_ignores_other_nodes(self):
+        """
+        Nodes in the configuration other than the one specified as the primary
+        for the new dataset have their configuration left alone by the
+        operation.
+        """
+        saving = self.persistence_service.save(Deployment(nodes={
+            Node(hostname=self.NODE_A)
+        }))
+
+        def saved(ignored):
+            return self.assertResponseCode(
+                b"POST", b"/datasets", {u"primary": self.NODE_B}, OK
+            )
+        saving.addCallback(saved)
+
+        def created(ignored):
+            deployment = self.persistence_service.get()
+            (node_a,) = (
```

Comment:
Or we should fix https://clusterhq.atlassian.net/browse/FLOC-266 somehow and start using it